### PR TITLE
Update testing to new approach fixes #18

### LINKED ISSUE #18: Update testing to new approach
We should update the testing to use Cloud Build and a new Service Account per run, as documented [here](https://docs.google.com/document/d/1eiHpZlhEDMDG8gi4LHyv2aZs1TJmeFlF8sqCE57_j4A/edit#heading=h.rz8l5qn0ygc0).

### PR DESCRIPTION
**Pull request by ingwarr**
Fixes #18

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,137 +12,74 @@
...
+# Enter docker container for local development
 .PHONY: docker_run
 docker_run:
 	docker run --rm -it \
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v "$(CURDIR)":/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
+		-e SERVICE_ACCOUNT_JSON \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: `$(CURDIR)` should be quoted, i.e. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 2]
```diff
@@ -12,137 +12,74 @@
...
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "kitchen create"
-
-.PHONY: docker_converge
-docker_converge:
+		-e SERVICE_ACCOUNT_JSON \
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: `$(CURDIR)` should be quoted, i.e. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 3]
```diff
@@ -12,137 +12,74 @@
...
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "kitchen converge"
-
-.PHONY: docker_verify
-docker_verify:
+		-e SERVICE_ACCOUNT_JSON \
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: `$(CURDIR)` should be quoted, i.e. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 4]
```diff
@@ -12,137 +12,74 @@
...
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v "$(CURDIR)":/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "kitchen verify"
-
-.PHONY: docker_destroy
-docker_destroy:
+		-e SERVICE_ACCOUNT_JSON \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: `$(CURDIR)` should be quoted, i.e. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 5]
```diff
@@ -12,137 +12,74 @@
...
+# Execute lint tests within the docker container
+.PHONY: docker_test_lint
+docker_test_lint:
 	docker run --rm -it \
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v "$(CURDIR)":/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "kitchen destroy"
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: `$(CURDIR)` should be quoted, i.e. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 6]
```diff
@@ -12,137 +12,74 @@
...
+		/usr/local/bin/test_lint.sh
 
-.PHONY: test_integration_docker
-test_integration_docker: docker_create docker_converge docker_verify docker_destroy
-	@echo "Running test-kitchen tests in docker"
+# Generate documentation
+.PHONY: docker_generate_docs
+docker_generate_docs:
+	docker run --rm -it \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: `$(CURDIR)` should be quoted, i.e. `-v "$(CURDIR)":/workspace`

File: test/setup/main.tf [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  billing_account   = var.billing_account
+
+  activate_apis = [
+    "cloudresourcemanager.googleapis.com",
+    "storage-api.googleapis.com",
+    "serviceusage.googleapis.com",
+    "compute.googleapis.com",
+  ]
+}
+
```
Comment (Jberlinsky) [inline test/setup/main.tf]: Unnecessary whitespace at the end of the file.

File: Makefile (hunk 3 above)
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile (hunk 2 above)
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile (hunk 1 above)
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile (hunk 4 above)
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile (hunk 5 above)
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile (hunk 6 above)
Comment (ingwarr) [reply Makefile]: Fixed

File: test/setup/main.tf (hunk 7 above)
Comment (ingwarr) [reply test/setup/main.tf]: Fixed

**Comment by kopachevsky** [comment]
@Jberlinsky 
> 1 files have incorrect boilerplate headers:
> test/source.sh

this file is temporary and ignored, usually it should not be there during lint test, so we can ignore it

**Comment by kopachevsky** [comment]
@Jberlinsky please do another review round
